### DISCUSSION TITLE
Image not showing properly

### DISCUSSION THREAD

**Discussion by Sinuelo** [TomSchimansky/CustomTkinter#2040] [Q&A]
Hello!

I'm working on a small project, a QR code generator, and i'm having some problems trying to display the image on the CTk window. 

![image](https://github.com/TomSchimansky/CustomTkinter/assets/98895433/3834b8e7-a2e2-4274-a6b1-de2582cd9ead)

![image](https://github.com/TomSchimansky/CustomTkinter/assets/98895433/d1929b41-3124-43d3-bed6-937138cfffeb)

If i try using ImageTk instead: 
```python
imagem = ImageTk.PhotoImage(Image.open(arquivo))
``` 
the code works but i get this message:
![image](https://github.com/TomSchimansky/CustomTkinter/assets/98895433/d32db4d0-e50c-49ed-9edb-35baa03a3337)

What am i doing wrong? 
Thanks in advance

**Comment by Akascape** [comment]
@Sinuelo 
Change the label text to " " and the image size.

```python
width, height = 100, 100
imagem = customtkinter.CTkImage(Image.open(arquivo).resize((width, height), Image.Resampling.LANCZOS))
```

Or 

```python
width, height = 100, 100
imagem = customtkinter.CTkImage(Image.open(arquivo), size=(width, height))
```

**Comment by Sinuelo** [comment]
It got rid of the text behind the QR code, however it's still small
![image](https://github.com/TomSchimansky/CustomTkinter/assets/98895433/2f86eb50-5a19-48ef-a440-ce671e70a9fa)

This is what i get when i try to use Image.Resampling.LANCZOS

![image](https://github.com/TomSchimansky/CustomTkinter/assets/98895433/77383fda-e133-4c28-9a0d-4103533ae93b)

Full code:
```python
def gerar_codigo():
    link = campo_link.get()
    nome_link = campo_nome.get()
    arquivo = nome_link + '.png'
    url = pyqrcode.create(link)
    url.png(arquivo, scale=8)
    width, height = 100, 100
    imagem = customtkinter.CTkImage(Image.open(arquivo).resize((width, height), Image.Resampling.LANCZOS))
    label_imagem = customtkinter.CTkLabel(janela, image=imagem, text='')
    label_imagem.image = imagem
    canvas.create_window(200, 430, window=label_imagem)


# Criando a janela
janela = CTk()
customtkinter.set_appearance_mode('light')

canvas = CTkCanvas(janela, width=400, height=600)
canvas.configure(bg='white')
canvas.pack()


# Criando o label de titulo
label_app = customtkinter.CTkLabel(janela, text='Gerador de Código QR', text_color='blue', fg_color='white', font=('Arial', 30, 'bold'))
canvas.create_window(200, 50, window=label_app)


# Criando o label nome e link
nome_label = customtkinter.CTkLabel(janela, text='Nome do Link', fg_color='white', text_color='black')
canvas.create_window(200, 100, window=nome_label)


link_label = customtkinter.CTkLabel(janela, text='Link', fg_color='white', text_color='black')
canvas.create_window(200, 160, window=link_label)


# Criando o Entry field do nome
campo_nome = customtkinter.CTkEntry(janela, placeholder_text_color='black', placeholder_text='Digite o nome do Link', fg_color='white', text_color='black', border_color='#2588E1')
canvas.create_window(200, 130, window=campo_nome)


# Criando o Entry field do link
campo_link = customtkinter.CTkEntry(janela, placeholder_text_color='black', placeholder_text='Cole o link aqui', fg_color='white', text_color='black', border_color='#2588E1')
canvas.create_window(200, 190, window=campo_link)


# Criando o botão
botao = customtkinter.CTkButton(janela, text='Gerar Código QR', command=gerar_codigo)
canvas.create_window(200, 235, window=botao)


janela.mainloop()

``` 
Thanks for taking the time to help me!

**Comment by Sinuelo** [comment]
While i have your attention: i've noticed that the color behind the QR code is grayish, it also happens if i use 
```python
corner_radius=20
``` 
![image](https://github.com/TomSchimansky/CustomTkinter/assets/98895433/568afb07-78be-4d56-bec7-bbe9ca1be80b)

When i set the theme to dark:
![image](https://github.com/TomSchimansky/CustomTkinter/assets/98895433/00d35ad0-b647-4bd4-8cda-41d5d197687d)

Do you know how i can fix this?
Thanks for your attention

**Comment by Akascape** [comment]
@Sinuelo First of all, why are you placing the widget inside a canvas? Why not use a CTkFrame? That is the reason you are not able to change the themes. Use the simple grid/pack methods to place the widgets inside a ctkframe.

Another method to change the size of image:
```python
imagem = customtkinter.CTkImage(Image.open(arquivo), size=(width, height))
```

And there is no need to add this line: `label_imagem.image = imagem`

**Reply by Sinuelo** [reply]
> @Sinuelo First of all, why are you placing the widget inside a canvas? Why not use a CTkFrame? That is the reason you are not able to change the themes. Use the simple grid/pack methods to place the widgets inside a ctkframe.
> 
> Another method to change the size of image:
> ```python
> imagem = customtkinter.CTkImage(Image.open(arquivo), size=(width, height))
> ```
> 
> And there is no need to add this line: `label_imagem.image = imagem`
> 

I understand your frustration, mate. I'm relatively new to this and I'm learning as I go. Thanks for helping though
